src/staging/src/components/advisories/advisoryFilter.js

```diff
@@ -0,0 +1,263 @@
+import React, { useState } from "react"
+import { makeStyles } from "@material-ui/core/styles"
+
+import {
+  Box,
+  Grid,
+  Button,
+  TextField,
+  InputAdornment,
+  Checkbox,
+  InputLabel,
+  FormControlLabel,
+  NativeSelect
+} from "@material-ui/core"
+
+import SearchIcon from "@material-ui/icons/Search"
+
+
+const useStyles = makeStyles(theme => ({
+  searchBox: {
+    marginRight: "10px",
+    width: "100%"
+  },
+  typeSelect: {
+    width: "100%",
+    borderRadius: "4px",
+    paddingLeft: "8px",
+    border: "solid 1px #aaa",
+    '&::before': {
+      borderBottom:0
+    }
+  },
+  filterElement: {
+    [theme.breakpoints.up('sm')]: {
+      marginRight: "4%",
+    },
+    [theme.breakpoints.down('sm')]: {
+      marginBottom: "15px",
+      marginRight: "10px",
+      "& .MuiCheckbox-root": {
+        marginRight: "0px",
+        paddingRight: "5px"
+      }      
+    },
+  },
+  filterSearch: {
+    [theme.breakpoints.down('sm')]: {
+      width: "99%"
+    },
+    [theme.breakpoints.up('sm')]: {
+      width: "25%"
+    }
+  },
+  filterType: {
+    [theme.breakpoints.down('sm')]: {
+      width: "60%"
+    },
+    [theme.breakpoints.up('sm')]: {
+      width: "15%"
+    }
+  },
+  filterLabel: {
+    fontWeight: "bold",
+    fontSize: "1rem",
+    lineHeight: "1.75rem",
+    color:"#000",
+    fontFamily: "unset",
+    marginBottom: "0px",
+  },
+  filterBy: {
+    [theme.breakpoints.up('sm')]: {
+    display: 'inline-flex',
+    marginRight: '15px',
+    verticalAlign: 'middle',
+    marginBottom: '10px',
+    },
+  }
+}))
+
+const AdvisoryFilter = ({ filterFunctions }) => { 
+
+  const classes = useStyles()
+
+  // Get parent's filter functions 
+  const getSearchText = filterFunctions.getSearchText;
+  const setSearchText = filterFunctions.setSearchText;
+  const setFilter = filterFunctions.setFilter;
+  const getFilter = filterFunctions.getFilter;
+  const getType = filterFunctions.getType;
+  const setType = filterFunctions.setType; 
+
+  const [filterText, setFilterText] = useState(
+    getSearchText() 
+  )
+  const [isParksFilter, setIsParksFilter] = useState(
+    getFilter("parks")
+  )
+  const [isTypesFilter, setIsTypesFilter] = useState(
+    getFilter("types")
+  )
+  const [isKeywordFilter, setIsKeywordsFilter] = useState(
+    getFilter("keyword")
+  )
+  const [advisoryType, setAdvisoryType] = useState(
+    getType()
+  )
+
+
+  // Local handlers, calls to parent methods
+  // will trigger useEffect functions in parent
+  const updateAdvisoriesSearchText = (str) => {
+   setSearchText(str) 
+  }
+  const handleSearch = () => {
+    setSearchText(filterText)
+  }
+
+  const handleTypeFilterChange = (str) => {
+    // This changes the URL query str and causes the page to
+    // rerender with the type changed
+    let newLoc = "/alerts/?type=" + str;
+    if (str !== "public") {
+      newLoc += "s"; // add "s" to wildfire and flood
+    }
+    window.location = newLoc; // this will trigger a reload
```

Comment:
I can see we want to update the URL with the new params but the reload feels pretty odd to me. We probably want to add [Reach router](https://www.gatsbyjs.com/docs/reach-router-and-gatsby/) or something so we can have cool URLs everywhere without reloads.

(future enhancement, not for this PR)

Reply:
totally agree a bit clunky, did the simplest thing, knowing that we want to put more params in url in future ticket